benchmark/definition/basic_benchmarking.py

```diff
@@ -0,0 +1,36 @@
+"""A basic benchmarking class for testing."""
+
+import time
+
+from attrs import define, field
+from typing_extensions import override
+
+from benchmark.definition.base import Benchmark
+from benchmark.result.basic_results import SingleResult
+
+
+@define
+class SingleExecutionBenchmark(Benchmark):
+    """A basic benchmarking class for testing a single benchmark execution."""
+
+    _benchmark_result: SingleResult = field(default=None)
```

Comment:
This is not the only place but I'm commenting only once: your types don't match. You can't have something like `x: SomethingNotNone` and then set its default to `None` (`mypy` will also complain about this). If this is necessary, you need to turn it into an optional type a la `SomethingNotNone | None`